api/src/main/java/com/example/api/util/csv/PointsToGradeMapper.java

```diff
@@ -24,4 +28,21 @@ public Double getGrade(double receivedPoints, double maxPoints) {
             return 5.0;
         }
     }
+
+    public Double getGrade(TaskResult taskResult) {
+        return getGrade(taskResult.getPointsReceived(), getMaxPoints(taskResult));
+    }
+
+    public Double getMaxPoints(TaskResult taskResult) {
```

Comment:
KIedyś używałeś instanceof i stwierdziłem, że git, ale jak już drugi raz to imo za dużo. Na szybko wygooglowałem to co myśle, że to bad habit jest no i stackoverflow potwierdziło. 

https://stackoverflow.com/questions/3776862/when-should-and-shouldnt-instanceof-be-used

Spróbuj to zrefactorować, żeby tego nie było.